File: Javascript/FindTheDifferenceOfTwoArrays.js
// Leetcode Problem - 2215

var findDifference = function(nums1,nums2){
    const output1 = []
    const output2 = []
    let outputArray = []

    let start1 = 0
    let start2 = 0

    const length1 = nums1.length
    const length2 = nums2.length

    while (start1<length1 && start2<length2){
        if (!nums2.includes(nums1[start1])){
            if (!output1.includes(nums1[start1])){
                output1.push(nums1[start1])
            }
        }

        if (!nums1.includes(nums2[start2])){
            if (!output2.includes(nums2[start2])){
                output2.push(nums2[start2])
            }
        }

        start1 += 1
        start2 += 1
    }

    while (start1<length1){
        if (!nums2.includes(nums1[start1])){
            if (!output1.includes(nums1[start1])){
                output1.push(nums1[start1])
            }
        }
        start1+=1
    }

    while (start2<length2){
        if (!nums1.includes(nums2[start2])){
            if (!output2.includes(nums2[start2])){
                output2.push(nums2[start2])
            }
        }
        start2+=1
    }

    outputArray = [output1,output2]

    return outputArray
}

File: Python/FindTheDifferenceOfTwoArrays.py
## Leetcode Problem - 2215

def findDifference(nums1,nums2):
    output1 = []
    output2 = []
    outputArray = []


    start1 = 0
    start2 = 0

    length1 = len(nums1)
    length2 = len(nums2)

    while start1<length1 and start2<length2:

        if nums1[start1] not in nums2:
            if nums1[start1] not in output1:
                output1.append(nums1[start1])

        if nums2[start2] not in nums1:
            if nums2[start2] not in output2:
                output2.append(nums2[start2])

        start1+=1
        start2+=1

    while start1<length1:
        if nums1[start1] not in nums2:
            if nums1[start1] not in output1:
                output1.append(nums1[start1])
        start1+=1

    while start2<length2:
        if nums2[start2] not in nums1:
            if nums2[start2] not in output2:
                output2.append(nums2[start2])
        start2+=1

    outputArray = [output1,output2]

    return outputArray


nums1 = [1,2,3]
nums2 = [2,4,6]

print(findDifference(nums1,nums2))